tools/toolchain/gen_mksurf_namelist.py

```diff
@@ -0,0 +1,658 @@
+        self.lu_fname = lu_fname
+
+    def create_landuse(self):
+        self.landuse_filename()
+        lu_file = open (self.lu_fname,'w')
+
+        for yr in tqdm.tqdm(range(self.start_year, self.end_year+1)):
+
+            #-- choose different files for years of 850-1850:
+            if (849 < yr < 1850):
+                lu_input_fname = os.path.join(self.input_path, 
+                            "pftcftdynharv.0.25x0.25.LUH2.histsimyr0850-1849.c171012",
+                            "mksrf_landuse_histclm50_LUH2_"+str(yr)+".c171012.nc"
+                            )
+            elif (1849 < yr < 2016): 
+                lu_input_fname = os.path.join(self.input_path, 
+                            "pftcftlandusedynharv.0.25x0.25.MODIS.simyr1850-2015.c170412",
+                            "mksrf_landuse_histclm50_LUH2_"+str(yr)+".c170412.nc"
+                            )
+            elif (2015 < yr < 2100 ):
+                self.decode_ssp_rcp()
+                lu_input_fname = os.path.join(self.input_path, 
+                            "pftcftdynharv.0.25x0.25."+self.ssp_rcp+".simyr2016-2100.c181217",
+                            "mksrf_landuse_SSP"+self.ssp_val+"RCP"+self.rcp_val+
+                            "_clm5_"+str(yr)+".c181217.nc")
+            else:
+                #--This error should never happen because
+                #  we have already checked years bound.
+                print ("year:", yr," not valid.")
+
+            #-- Check if the land-use input file exist:
+            if not os.path.isfile(lu_input_fname):
+                print ("lu_input_fname:", lu_input_fname)
+                print ("WARNING: land-use input file does not exist for year:", yr,".")
+
+            # TODO: make the space/tab exactly the same as pl code:
+            lu_line = lu_input_fname + "\t\t\t"+ str(yr) + "\n"
+
+            #-- Each line is written twice in the original pl code:
+            lu_file.write (lu_line)
+            lu_file.write (lu_line)
+
+            logging.debug('year : %s', yr)
+            logging.debug(lu_line)
+
+        print ("Successfully created land use file : ", self.lu_fname,".")
+        print ("-------------------------------------------------------")
+
+
+
+    def decode_ssp_rcp(self):
+        """
+        Decode ssp_rcp strings because
+        the raw filenames and folder names 
+        are not consistent.
+
+        For example:
+        folder names have ssp_rcp as SSP1-2.6
+
+        but the files in these folders have 
+        ssp_rcp as SSP1RCP26
+        """
+
+        if self.ssp_rcp is not "hist":
+            temp = re.sub("[^0-9]", "",self.ssp_rcp)
+            self.ssp_val = temp[0]
+            self.rcp_val = temp[1:]
+        else:
+            sys.exit('ERROR: \n'+
+                 '\t Please choose a ssp_rcp scenario for years beyond 2015 using --ssp_rcp flag.')
+
+
+    def build_nl (self):
+        """
+        Build the namelist/control file for a ctsm class.  
```

Comment:
```suggestion
        Build the namelist/control file for a ctsm case.
```
...though elsewhere I recommended removing the words "ctsm case" altogether.

Reply:
I changed to the ctsm case here.